### PR TITLE
GRAD2-1519 & GRAD2-1722 : the tasks are complete.

### PR DESCRIPTION
**Pull request by infstar**
GRAD2-1519 & GRAD2-1722 : the tasks are complete.

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/api/batchgraduation/listener/DistributionRunCompletionNotificationListener.java [hunk 1]
```diff
@@ -59,6 +62,27 @@ public void afterJob(JobExecution jobExecution) {
...
 
+	private String populateJobParametersDTO(String jobType, String studentSearchRequest) {
+		JobParametersForDistribution jobParamsDto = new JobParametersForDistribution();
+		jobParamsDto.setJobName(jobType);
+
+		try {
+			StudentSearchRequest payload = new ObjectMapper().readValue(studentSearchRequest, StudentSearchRequest.class);
+			jobParamsDto.setPayload(payload);
+		} catch (Exception e) {
+			e.printStackTrace();
```
Comment (cditcher) [inline api/src/main/java/ca/bc/gov/educ/api/batchgraduation/listener/DistributionRunCompletionNotificationListener.java:73]: Log error

**Review by cditcher — CHANGES_REQUESTED** [review]
populateJobParametersDTO is so similar in listener classes, can we make a generic method based on the second parameter 'transmissionType or credentialType, etc?)

File: api/src/main/java/ca/bc/gov/educ/api/batchgraduation/listener/DistributionRunCompletionNotificationListener.java (hunk 1 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/batchgraduation/listener/DistributionRunCompletionNotificationListener.java]: done

**Comment by infstar** [comment]
> populateJobParametersDTO is so similar in listener classes, can we make a generic method based on the second parameter 'transmissionType or credentialType, etc?)

Refactoring is done.

**Review by cditcher — APPROVED** [review]

**Pull request merged**